test/snippet/search/algorithm/search.cpp

```diff
@@ -0,0 +1,28 @@
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/core/debug_stream.hpp>
+#include <seqan3/search/algorithm/search.hpp>
+#include <seqan3/search/fm_index/all.hpp>
+
+int main()
+{
+    using seqan3::operator""_dna4;
+    std::vector<std::vector<seqan3::dna4>> genomes{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTA"_dna4,
+                                                   "ACCCGATGAGCTACCCAGTAGTCGAACTG"_dna4,
+                                                   "GGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
+    std::vector<std::vector<seqan3::dna4>> queries{"GCT"_dna4, "ACCC"_dna4};
+
+    // build an index, here a bidirectional FM index is used, but any other index SeqAn offers is possible
```

Comment:
This is really good to provide some extra information on that
But I am afraid that according to the top-level documentation the sentence is not completely true. It says that the index must be model `seqan3::fm_index_specialisation`? In that case I suggest using the standard seqan3::fm_index.
But honestly, the entire search algorithm documentation would need some more love I guess. Let's discuss the open issues in Monday

Reply:
Alright, sorry @MitraDarja then remove the comment for now. 

I still think this snippet will already greatly improve the documentation while we sort out who puts more love in it.